test/models/subscription_status_test.rb

```diff
@@ -37,10 +41,16 @@ def test_to_status_invalid
     end
 
     def test_to_status_unsubscribed_hypervisor
-      stub_status('unsubscribed_hypervisor')
+      host.subscription_facet.stubs(:unsubscribed_hypervisor?).returns(true)
       assert_equal Katello::SubscriptionStatus::UNSUBSCRIBED_HYPERVISOR, status.to_status
     end
 
+    def test_sca_enabled_hypervisor
+      host.subscription_facet.stubs(:unsubscribed_hypervisor?).returns(false)
```

Comment:
If I am understanding the BZ properly, we need to show DISABLED status for hypervisors in SCA mode. So the assertion for this test is correct but the setup is wrong . The unsubscribed_hypervisor? stub should be returning true here. Does that sound right?

In which case I think the below stub_status could be eliminated, and if not let's understand why.

Reply:
Ah, right. unsubscribed_hypervisor itself checks for sca mode. Maybe that should be removed and then the status class can do this:

```
status_override = Katello::Candlepin::Consumer::ENTITLEMENTS_DISABLED if host.organization.simple_content_access?
status_override ||= 'unsubscribed_hypervisor' if host.subscription_facet.unsubscribed_hypervisor?
status_override ||= options.fetch(:status_override, nil)
status = status_override || host.subscription_facet.candlepin_consumer.entitlement_status
```

That will make the test case more clear so unsubscribed_hypervisor can be stubbed to true which illustrates the requirement. Don't think that will mess with any tests you've encountered

Reply:
Let me look at this code a bit more. I think we need a small adjustment.

Reply:
This method probably needs a bit of a refactor, but let's keep it simple for now. Try (and test) my suggestion above